tests/test_diffraction_DiffractionPattern.py

```diff
@@ -159,76 +166,83 @@ def test_repr(self):
+        # np.fft.fftfreq and the _k_scale_factor to write an expression for
+        # the first and last bins' k values and k vectors.
+
+        scale_factor = 2 * np.pi * output_size / (np.max(box.to_matrix()) * zoom)
+
+        if output_size % 2 == 0:
+            first_k_value = -0.5 * scale_factor
+            last_k_value = (0.5 - (1 / output_size)) * scale_factor
+        else:
+            first_k_value = (-0.5 + 1 / (2 * output_size)) * scale_factor
+            last_k_value = (0.5 - 1 / (2 * output_size)) * scale_factor
+
+        npt.assert_allclose(dp.k_values[0], first_k_value)
+        npt.assert_allclose(dp.k_values[-1], last_k_value)
+
+        first_k_vector = rowan.rotate(
+            view_orientation, [first_k_value, first_k_value, 0]
+        )
+        last_k_vector = rowan.rotate(view_orientation, [last_k_value, last_k_value, 0])
+        top_right_k_vector = rowan.rotate(
+            view_orientation, [first_k_value, last_k_value, 0]
+        )
+        bottom_left_k_vector = rowan.rotate(
+            view_orientation, [last_k_value, first_k_value, 0]
+        )
+
+        npt.assert_allclose(dp.k_vectors[0, 0], first_k_vector)
+        npt.assert_allclose(dp.k_vectors[-1, -1], last_k_vector)
+        npt.assert_allclose(dp.k_vectors[0, -1], top_right_k_vector)
+        npt.assert_allclose(dp.k_vectors[-1, 0], bottom_left_k_vector)
+
+        center = output_size // 2
+        top_center_k_vector = rowan.rotate(view_orientation, [0, first_k_value, 0])
+        bottom_center_k_vector = rowan.rotate(view_orientation, [0, last_k_value, 0])
+        left_center_k_vector = rowan.rotate(view_orientation, [first_k_value, 0, 0])
+        right_center_k_vector = rowan.rotate(view_orientation, [last_k_value, 0, 0])
+
+        npt.assert_allclose(dp.k_vectors[center, 0], top_center_k_vector)
+        npt.assert_allclose(dp.k_vectors[center, -1], bottom_center_k_vector)
+        npt.assert_allclose(dp.k_vectors[0, center], left_center_k_vector)
+        npt.assert_allclose(dp.k_vectors[-1, center], right_center_k_vector)
+
+    @pytest.mark.parametrize(
+        "grid_size, output_size, zoom",
+        [
+            (grid_size, output_size, zoom)
+            for grid_size in (256, 1024)
+            for output_size in (255, 256, 1023, 1024)
+            for zoom in (1, 2.5, 4.123)
+        ],
+    )
+    def test_cubic_system(self, grid_size, output_size, zoom):
         length = 1
         box, positions = freud.data.UnitCell.sc().generate_system(
             num_replicas=16, scale=length, sigma_noise=0.1 * length
         )
         # Pick a non-integer value for zoom, to ensure that peaks besides k=0
         # are not perfectly aligned on pixels.
-        dp = freud.diffraction.DiffractionPattern(grid_size=512)
-        dp.compute((box, positions), zoom=4.123)
+        dp = freud.diffraction.DiffractionPattern(
+            grid_size=grid_size, output_size=output_size
```

Comment:
One of the cases that is no longer explicitly covered here is the _default_ behavior of `output_size` when no value is provided. I don't recall exactly how that works but I would like to ensure that case is still tested.